account_payment_term_sale_purchase_restriction/models/account_payment_term.py

```diff
@@ -0,0 +1,31 @@
+from odoo import api, fields, models
+
+
+class AccountPaymentTerm(models.Model):
+    _inherit = "account.payment.term"
+
+    apply_on = fields.Selection(
+        selection="_selection_apply_on",
+        string="Applicable On",
+        default=lambda self: self._get_default_apply_on(),
+    )
+
+    @api.model
+    def _selection_apply_on(self):
+        return [
+            ("sale", "Sales"),
+            ("purchase", "Purchase"),
+            ("both", "Both"),
```

Comment:
```suggestion
            ("all", "All"),
```
Maybe we could rename this so that it does not look weird if other modules modify the selection?